subprojects/launcher/src/main/java/org/gradle/tooling/internal/provider/ContinuousBuildActionExecuter.java

```diff
@@ -70,30 +70,34 @@ public ContinuousBuildActionExecuter(BuildActionExecuter<BuildActionParameters>
     @Override
     public Object execute(BuildAction action, BuildRequestContext requestContext, final BuildActionParameters actionParameters, ServiceRegistry buildSessionScopeServices) {
         BuildCancellationToken cancellationToken = requestContext.getCancellationToken();
-
-        final CancellableOperationManager cancellableOperationManager;
-        if (actionParameters.isInteractive()) {
-            if (!(System.in instanceof DisconnectableInputStream)) {
-                System.setIn(new DisconnectableInputStream(System.in));
-            }
-            DisconnectableInputStream inputStream = (DisconnectableInputStream) System.in;
-            cancellableOperationManager = new DefaultCancellableOperationManager(executorFactory.create("cancel signal monitor"), inputStream, cancellationToken);
-        } else {
-            cancellableOperationManager = new PassThruCancellableOperationManager(cancellationToken);
-        }
-
         if (actionParameters.isContinuous()) {
             SingleMessageLogger.incubatingFeatureUsed("Continuous build");
             DefaultContinuousExecutionGate alwaysOpenExecutionGate = new DefaultContinuousExecutionGate();
+            final CancellableOperationManager cancellableOperationManager = createCancellableOperationManager(actionParameters, cancellationToken);
             return executeMultipleBuilds(action, requestContext, actionParameters, buildSessionScopeServices, cancellableOperationManager, alwaysOpenExecutionGate);
         } else {
             try {
                 return delegate.execute(action, requestContext, actionParameters, buildSessionScopeServices);
             } finally {
+                final CancellableOperationManager cancellableOperationManager = createCancellableOperationManager(actionParameters, cancellationToken);
```

Comment:
Can you explain why we also create a `CancellableOperationManager` here?

Reply:
If we detect running deployments at the end of the build when we are _not_ in continuous build, we enter a sort of 'continuous build' after the first build.  It needs to be cancellable via ctrl+C, so it needs a `CancellableOperationManager`